nats.go

```diff
@@ -261,6 +269,14 @@ type Options struct {
 	// dictated by PendingLimits()
 	SubChanLen int
 
+	// Nkey sets the public nkey that will be used to authenticate
+	// when connecting to the server
+	Nkey string
+
+	// SignatureCB designates the function used to sign the nonce
+	// presented from the server.
+	SignatureCB SignatureHandler
```

Comment:
Maybe we could use a small `keypair` interface for signing the nonces as an alternative to the callback, for example:

```
diff --git a/nats.go b/nats.go
index 9eaa71e..71b4d04 100644
--- a/nats.go
+++ b/nats.go
@@ -168,6 +168,11 @@ type CustomDialer interface {
 	Dial(network, address string) (net.Conn, error)
 }
 
+type keypair interface {
+	Sign(input []byte) ([]byte, error)
+	PublicKey() (string, error)
+}
+
 // Options can be used to create a customized connection.
 type Options struct {
 
@@ -277,6 +282,9 @@ type Options struct {
 	// presented from the server.
 	SignatureCB SignatureHandler
 
+	// Nkeys is the keypair that will be used to sign nonces sent on connect.
+	Nkeys keypair
+
```

That way we could support usage like this:

```go
	kp, err := nkeys.FromSeed("SUAEL6....")
	if err != nil {
		log.Fatal(err)
	}
	opts := nats.DefaultOptions
	opts.Nkeys = kp

	nc, err := opts.Connect()
	if err != nil {
		log.Fatal(err)
	}
```

Internally the client would be the one doing the signing, but since using the small interface does not require to import "github.com/nats-io/nkeys" to be able to do it.

```
@@ -1297,12 +1300,21 @@ func (nc *Conn) connectProto() (string, error) {
 		user = nc.Opts.User
 		pass = nc.Opts.Password
 		token = nc.Opts.Token
		nkey = nc.Opts.Nkey
+		nkeys = nc.Opts.Nkeys
 	}
 
+	if nkeys != nil {
+		nonce := []byte(nc.info.Nonce)
+		signed, err := nkeys.Sign(nonce)
+		if err != nil {
+			return _EMPTY_, err
+		}
+		sig = base64.StdEncoding.EncodeToString(signed)
+
+		nkey, err = nkeys.PublicKey()
+		if err != nil {
+			return _EMPTY_, err
+		}
```

Reply:
I think that would actually end up making more work for the user even if our api looked simpler.